### PR TITLE
Add new cask TextSniper v1.7.0

### PR DESCRIPTION
**Pull request by noranraskin**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by noranraskin** [comment]
Two nice additions would be to add a livecheck and to close the running application when uninstalling.
However, I do not know how to do this. It uninstalls successfully nonetheless.

File: Casks/textsniper.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+cask "textsniper" do
+  version "1.7.0"
+  sha256 :no_check
+
+  url "https://textsniper.app/api/downloads/mac-latest"
```
Comment (neutric) [inline Casks/textsniper.rb:5]:
```
% curl -I https://textsniper.app/api/downloads/mac-latest
HTTP/2 307 
location: https://s3.amazonaws.com/textsniper.app/TextSniper1.7.0.dmg
```
Please switch to versioned `url` which allows for a checksum.

File: Casks/textsniper.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  name "textsniper"
+  desc "Extract text from images and other digital documents in seconds"
+  homepage "https://textsniper.app/"
+
+  auto_updates true
+
+  app "TextSniper.app"
+end
```
Comment (neutric) [inline Casks/textsniper.rb:13]: Please try to add a [`zap` stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap). The linked documentation provides some recommended methods to create it.

File: Casks/textsniper.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+cask "textsniper" do
+  version "1.7.0"
+  sha256 :no_check
+
+  url "https://textsniper.app/api/downloads/mac-latest"
+  name "textsniper"
+  desc "Extract text from images and other digital documents in seconds"
+  homepage "https://textsniper.app/"
+
+  auto_updates true
```
Comment (neutric) [inline Casks/textsniper.rb:10]:
```
livecheck do
  url "https://textsniper.app/api/downloads/mac-latest"
  strategy :header_match
end
```

**Review by neutric — COMMENTED** [review]
For quitting running apps prior to uninstalling, check out the [`uninstall` stanza](https://docs.brew.sh/Cask-Cookbook#stanza-uninstall).

File: Casks/textsniper.rb (hunk 3 above)
Comment (neutric) [reply Casks/textsniper.rb:10]: Also, `depends_on macos: ">= :catalina"` would be nice.

**Comment by noranraskin** [comment]
I don't know why I had to merge master into textsniper (I didn't modify iterm2) but I took the changes from master not from HEAD so that should solve the issue.
Also, I've added the uninstall and zap stanzas to textsniper, tested thoroughly and works like a charm.
@miccal You've changed it since, but how did you know "https://textsniper.app/latest_releases.xml" would give you back the releases and not just a 404? Just asking out of curiosity

**Comment by miccal** [comment]
> @miccal You've changed it since, but how did you know "https://textsniper.app/latest_releases.xml" would give you back the releases and not just a 404? Just asking out of curiosity

I looked inside the `plist` file inside the `app` bundle and found this in the `SUFeedURL` for software updates.

**Comment by noranraskin** [comment]
Wow thanks learned something again

**Pull request merged**